src/nm-ssh-service.c

```diff
@@ -515,7 +515,7 @@ send_network_config (NMSshPlugin *plugin)
 
 	/* replace default route? */
 	if (io_data->no_default_route) {
-		g_hash_table_insert (config, NM_VPN_PLUGIN_IP4_CONFIG_NEVER_DEFAULT, bool_to_gvalue (TRUE));
+		g_hash_table_insert (ip4config, NM_VPN_PLUGIN_IP4_CONFIG_NEVER_DEFAULT, bool_to_gvalue (TRUE));
```

Comment:
Or actually the attribute corresponds just to ipv4. I'll have to take a deeper look into that. Does it work for you?